src/Nox.Types/Types/User/User.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Nox.TypeOptions;
+using Nox.Enums;
+using System;
+using YamlDotNet.Core.Tokens;
+
+namespace Nox.Types;
+
+/// <summary>
+/// The user nox type.
+/// </summary>
+public sealed class User : ValueObject<string, User>
+{
+    private UserTypeOptions _userTypeOptions = new();
+
+    /// <summary>
+    /// Gets or sets the value.
+    /// </summary>
+    public override string Value { protected set => base.Value = value.ToLowerInvariant(); }
```

Comment:
not sure about the implications of this ToLowerInvariant

Reply:
The user id needs to be case insensitive for comparison. I don't like storing things in upper case here. So to lower invariant primarily on English characters should work, unless we are allowing Unicode characters in the username. That's certainly something worth considering.